### PR TITLE
Integrating PartnerIDs 

### PR DESCRIPTION
**Pull request by mtrinh11**

### CODE REVIEW

File: transport.go [hunk 1]
```diff
@@ -98,9 +103,39 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
 
 		wv.setWebhookDefaults(&webhook, r.RemoteAddr)
 
+		auth, _ := bascule.FromContext(c)
+		var partners []string
+
+		switch auth.Token.Type() {
+		case "Basic":
```
Comment (kristinapathak) [inline transport.go:110]:
```suggestion
		case "basic":
```

File: transport.go [hunk 2]
```diff
@@ -98,9 +103,39 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
...
+			var configurableString string
```
Comment (kristinapathak) [inline transport.go:111]: We can add this string (I would name it something like `basicPartnerIDsHeader` to the `transportConfig` that we pass into the create decoder function.  Then, if it's empty, we should set it to a const called `DefaultBasicPartnerIDsHeader`.  I would add this empty string check above the creation of this function to return (ie around line 82).

File: transport.go [hunk 3]
```diff
@@ -98,9 +103,39 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
...
+			for _, value := range authHeader {
+				fields := strings.Split(value, ",")
+				for i := 0; i < len(fields); i++ {
+					fields[i] = strings.TrimSpace(fields[i])
+				}
+				partners = append(partners, fields...)
+			}
+		case "jwt":
+			authToken := auth.Token
+			whatdoesthisinterfacemean, _ := bascule.GetNestedAttribute(authToken.Attributes(), basculechecks.PartnerKeys()...)
```
Comment (kristinapathak) [inline transport.go:123]: Might want to rename `whatdoesthisinterfacemean` 🙂

File: transport.go [hunk 4]
```diff
@@ -98,9 +103,39 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
...
+			vals, err := cast.ToStringSliceE(whatdoesthisinterfacemean)
+			if err != nil {
+				return nil, errGettingPartnerIDs
```
Comment (kristinapathak) [inline transport.go:126]:
```suggestion
				return nil, fmt.Errorf("%w: %v", errGettingPartnerIDs, err)
```
Usually we don't want to lose any error information.

**Review by kristinapathak — COMMENTED** [review]
Looking good so far!

File: transport.go [hunk 5]
```diff
@@ -98,9 +113,47 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
...
+				return nil, errPartnerIDsDoNotExist
+			}
+			vals, err := cast.ToStringSliceE(partnersInterface)
+			if err != nil {
+				return nil, fmt.Errorf("%w: %v", errGettingPartnerIDs, err)
+			}
+			partners = vals
+		default:
+			return nil, errAuthIsNotOfTypeBasicOrJWT
+		}
```
Comment (joe94) [inline transport.go:149]: we could make this its own helper function. Something like `extractPartnerIDs` which returns both a slice and an error type?

File: transport_test.go [hunk 6]
```diff
@@ -220,24 +274,60 @@ func TestAddWebhookRequestDecoder(t *testing.T) {
...
+				auth = bascule.Authentication{
+					Token: bascule.NewToken("jwt", "owner-from-auth", bascule.NewAttributes(
+						map[string]interface{}{"allowedResources": map[string]interface{}{"allowedPartners": nil}})),
+				}
+			case "authnotbasicorjwt":
+				auth = bascule.Authentication{
+					Token: bascule.NewToken("spongebob", "owner-from-auth", bascule.NewAttributes(
+						map[string]interface{}{})),
+				}
+			default:
```
Comment (joe94) [inline transport_test.go:305]: Go doesn't require the default case to be explicit when empty

**Review by joe94 — CHANGES_REQUESTED** [review]

File: transport.go [hunk 7]
```diff
@@ -41,7 +41,10 @@ var (
 	errFailedWebhookUnmarshal    = errors.New("failed to JSON unmarshal webhook")
 	errAuthIsNotOfTypeBasicOrJWT = errors.New("auth is not of type Basic of JWT")
 	errGettingPartnerIDs         = errors.New("unable to retrieve PartnerIDs")
-	DefaultBasicPartnerIDsHeader = ""
+	errAuthNotPresent            = errors.New("auth not present")
+	errAuthTokenIsNil            = errors.New("auth token is nil")
+	errPartnerIDsDoNotExist      = errors.New("partnerIDs do not exist")
+	DefaultBasicPartnerIDsHeader = "X-Xmidt-Partner-Id"
```
Comment (kristinapathak) [inline transport.go:47]:
```suggestion
	DefaultBasicPartnerIDsHeader = "X-Xmidt-Partner-Ids"
```

**Review by kristinapathak — APPROVED** [review]
Looks great!

**Review by joe94 — APPROVED** [review]
LGTM!

**Pull request merged**